extensions/okhttp/src/main/java/com/google/android/exoplayer2/ext/okhttp/OkHttpDataSource.java

```diff
@@ -280,7 +276,6 @@ private Request makeRequest(DataSpec dataSpec) {
       }
       builder.addHeader("Range", rangeRequest);
     }
-    builder.addHeader("User-Agent", userAgent);
```

Comment:
What about for the use case I described (it seems pretty reasonable to me that an app might want to include info about the ExoPlayer version in the user agent being sent in requests made by the player)? You could probably argue that you should just use a separate `OkHttpClient` instance. I'm not sure what the implications of that are, but it feels at least possible that there might be some efficiency loss there (e.g. if the okhttp implementation uses a separate connection pool for each client).

In any case, let's go for the less invasive change where we just make the user agent optional, in the `dev-v2` branch. Let me know if you plan to send such a change, or alternatively whether you'd like us to push one directly.

Thanks for bringing the issue to our attention :)!